packages/ember-routing/lib/system/router.js

```diff
@@ -865,6 +877,10 @@ function triggerEvent(handlerInfos, ignoreFailure, args) {
       if (handler.actions[name].apply(handler, args) === true) {
         eventWasHandled = true;
       } else {
+        // Should only hit here if a non-bubbling error action is triggered on a route.
+        if (name === 'error') {
+          markErrorAsHandled(args[0]);
```

Comment:
It isn't necessary to make it a variable, but I do not think that the following does what we want:

```js
e1 = new Error('foo');
e2 = new Error('foo');
obj = {};
obj[e1] = true;
obj[e2] = false;

obj[e1] // => false WAT
```

![screenshot](https://monosnap.com/file/E9AMczZANwfPn7OioLoWxqI3Mme4rI.png)